### PR TITLE
cosmosc2 stream client

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: cosmosc2/stream_api/base_client.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# vim: tabstop=8 expandtab shiftwidth=4 softtabstop=4
+# -*- coding: latin-1 -*-
+"""
+data_extractor_client.py
```
Comment (ryanatball) [inline cosmosc2/stream_api/base_client.py:5]: base_client.py

File: cosmosc2/stream.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+            logging.exception(e)
+        finally:
+            logging.debug(f"exitting task: {endpoint}")
+            self._queues.pop(endpoint, None)
+            self._events.pop(endpoint, None)
+
+    @staticmethod
+    async def _welcome(ws):
+        """
+        Confrim welcome from ActionCable websocket.
```
Comment (ryanatball) [inline cosmosc2/stream.py:166]: Confirm typo

File: cosmosc2/stream_api/base_client.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+from datetime import datetime
+import logging
+from threading import Event
+import time
+
+
+class BaseClient:
+    """
+    The BaseClient is designed to be a parent class for websocket
+    implamentations to expand upon.
```
Comment (ryanatball) [inline cosmosc2/stream_api/base_client.py:25]: implementations typo

File: cosmosc2/stream_api/log_message_client.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# vim: tabstop=8 expandtab shiftwidth=4 softtabstop=4
+# -*- coding: latin-1 -*-
+"""
+data_extractor_client.py
```
Comment (ryanatball) [inline cosmosc2/stream_api/log_message_client.py:5]: log_message_client.py

File: cosmosc2/stream_api/log_message_client.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        typ = message.get("type")
+        if self._count == self.count:
+            self._event.set()
+        elif typ is None and msg is not None:
+            self._last_msg = datetime.now().timestamp()
+            self._split_data(msg)
+
+    def get(self):
+        """
+        Get the data from the DataExtractor websocket.
```
Comment (ryanatball) [inline cosmosc2/stream_api/log_message_client.py:84]: LogMessage not DataExtractor

File: cosmosc2/stream_shared.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+from cosmosc2.stream import CosmosAsyncStream
+
+
+class CosmosAsyncClient:
+    def __init__(
+        self,
+        stream: CosmosAsyncStream,
+        scope: str = COSMOS_SCOPE,
+        auth: CosmosAuthorization = None,
+    ) -> None:
```
Comment (ryanatball) [inline cosmosc2/stream_shared.py:29]: What does -> None mean?

File: examples/cosmos_v5_stream_example.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+cosmos_v5_stream_example.py
+"""
+import argparse
+import hashlib
+import os
+import logging
+
+# See cosmosc2/docs/environment.md for environment documentation
+
+os.environ["COSMOS_VERSION"] = "1.1.1"
```
Comment (ryanatball) [inline examples/cosmos_v5_stream_example.py:14]: Hopefully this variable isn't required.

File: examples/cosmos_v5_stream_example.py [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+        for d in data:
+            f.write(f"{d['item']},{d['value']},{d['time']}\n")
+    print(filename)
+
+
+# item example: INST.ADCS.POSX
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "mode", type=str, choices=["CONVERTED", "DECOM", "RAW"], help="item mode"
```
Comment (ryanatball) [inline examples/cosmos_v5_stream_example.py:46]: Should also support WITH_UNITS

**Review by ghost — CHANGES_REQUESTED** [review]
Mainly typos.  Looks good overall.

File: cosmosc2/stream_shared.py (hunk 6 above)
Comment (gerhardatball) [reply cosmosc2/stream_shared.py:29]: returns nothing, way you can hint at what is being returned

File: examples/cosmos_v5_stream_example.py (hunk 7 above)
Comment (gerhardatball) [reply examples/cosmos_v5_stream_example.py:14]: No that allows you to track a version in the requests.

**Review by ghost — APPROVED** [review]

**Pull request merged**